Firmware/eeprom.cpp

```diff
@@ -191,7 +199,17 @@ void __attribute__((noinline)) eeprom_init_default_dword(uint32_t *__p, uint32_t
         eeprom_write_dword(__p, def);
 }
 
+void __attribute__((noinline)) eeprom_init_default_float(float *__p, float def) {
+    if (eeprom_read_dword((uint32_t*)__p) == EEPROM_EMPTY_VALUE32)
+        eeprom_write_float(__p, def);
```

Comment:
Do we need eeprom_write_float at all? Or was the AVR libc smart enough to implement it the same way like eeprom_write_dword?

Reply:
I do not know if it got implemented more efficiently. In theory the code is a duplicate of the above, but I hope the compiler doesn't treat uint32_t and float differently in this case

Reply:
I just checked the avr libc source and it does indeed look like the DWORD and FLOAT implementations in ASM are identical. Both symbols are attached to the same piece of code. As such, even the rest of the function should be identical. I expect the LTO to combine our two functions into one.

https://github.com/avrdudes/avr-libc/blob/55e8cac69935657bcd3e4d938750960c757844c3/libc/misc/eewr_dword.S#L45-L46